src/middleware.ts

```diff
@@ -0,0 +1,16 @@
+import { clerkMiddleware, createRouteMatcher } from '@clerk/nextjs/server'
+
+const isProtectedRoute = createRouteMatcher(['/dashboard(.*)'])
```

Comment:
Basically, any route that starts with `/dashboard` will require user to be authenticated, and if they are not, it will automatically re-route them to the sign in page